### PR TITLE
docs(dropdown): usage and style updates

### LINKED ISSUE #2123: [dropdown] updates to usage and style guidance
#### Usage tab

- [x] Combobox: Edit out all guidance about adding unique entries. Our combobox currently does not have this functionality. We can update this again once #[6102](https://github.com/carbon-design-system/carbon/issues/6102) is added to code.

- [x] Dropdown: Remove `optional` when talking about visual labels.
- [x] Dropdown: Split up label and helper-text under the anatomy section to expand more on our guidance.

#### Style tab
- [x] Update focus states for open menus. There should be a 1px focus on the open menu container and not on individual items.

### PR DESCRIPTION
**Pull request by laurenmrice**
Closes #2123 

Update usage and style guidance to reflect our current functionality.

----

**Changed**

Usage tab:

- Anatomy:
   - Gave "Label" and "Helper text" their own line items to explain them more clearly.
   - Removed `optional` text for visual labels.
   - Updated anatomy image to reflect new numbered list items.

- Combo box:
   - Changed content guidance to reflect the current functionality of combo box.
   - Removed guidance about adding unique entries. (for now)

-----

Style tab:

- Images:
   - Edited images that had open menus with focus.
   - Updated focus to a 1px focus border on the menu container instead of individual options.

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
looks good to me

File: src/pages/components/dropdown/usage.mdx [hunk 1]
```diff
@@ -705,8 +707,9 @@ While the dropdown and combo box look similar they have different functions.
 
 - With a dropdown list, the selected option is always visible, and the other
   options are visible by clicking and opening the list.
-- A combo box is a combination of a standard list box or a dropdown list and an
-  editable text box and users are able to enter a value that isn't in the list.
+- A combo box is a combination of a standard list box or a dropdown list that
+  lets you type inside the input field to find an option that best represents
```
Comment (janchild) [inline src/pages/components/dropdown/usage.mdx:711]:
```suggestion
  lets users type inside the input field to find an option that matches
```

File: src/pages/components/dropdown/usage.mdx [hunk 2]
```diff
@@ -705,8 +707,9 @@ While the dropdown and combo box look similar they have different functions.
...
+  your inputted value.
```
Comment (janchild) [inline src/pages/components/dropdown/usage.mdx:712]:
```suggestion
  their inputted value.
```

**Review by janchild — COMMENTED** [review]
@laurenmrice These are very nice additions! Just a few small copy tweaks. 🎉

**Review by janchild — APPROVED** [review]
💥

**Review by aagonzales — APPROVED** [review]
Looks good to me

**Pull request merged**